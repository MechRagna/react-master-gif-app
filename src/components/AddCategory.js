import React, { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState("");

  const handleInputChange = (event) => {
    setInputValue(event.target.value);
  };

  const handleSubmit = (event) => {
    event.preventDefault();

    if (inputValue.trim().length > 3) {
      setCategories((cats) => {
        if (!cats.includes(inputValue.toLowerCase())) {
          return [inputValue.toLowerCase(), ...cats];
        } else {
          return cats;
        }
      });
      setInputValue("");
    }
  };

  return (
    <form onSubmit={handleSubmit}>
      <input type="text" value={inputValue} onChange={handleInputChange} />
    </form>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};
