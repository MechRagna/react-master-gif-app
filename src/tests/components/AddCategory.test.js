import React from "react";
import { shallow } from "enzyme";
import { AddCategory } from "../../components/AddCategory";
import "@testing-library/jest-dom";

describe("Probar AddCategory", () => {
  const setCategories = jest.fn();
  let wrapper = shallow(<AddCategory setCategories={setCategories} />);

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<AddCategory setCategories={setCategories} />);
  });

  test("Tiene que matchear con el snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe cambiar la caja de texto", () => {
    const input = wrapper.find("input");
    const event = { target: { value: "One Punch" } };
    input.simulate("change", event);
  });

  test("NO debe postear el handleSubmit", () => {
    const event = { preventDefault() {} };
    wrapper.find("form").simulate("submit", event);
    expect(setCategories).not.toHaveBeenCalled();
  });

  test("Debe llamar al setCategories y limpiar la caja de texto ", () => {
    const eventChg = { target: { value: "One Punch" } };
    const input = wrapper.find("input");
    input.simulate("change", eventChg);
    const eventSubmit = { preventDefault() {} };
    wrapper.find("form").simulate("submit", eventSubmit);
    expect(setCategories).toHaveBeenCalled();
    expect(input.props().value).toBe("");
  });
});
