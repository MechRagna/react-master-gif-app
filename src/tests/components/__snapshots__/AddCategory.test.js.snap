// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probar AddCategory Tiene que matchear con el snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
